src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -0,0 +1,1062 @@
+            }
+
+            // The rest is for entire container aging.
+            // We must get at least one free slot.
+
+            // Don't need to apply LRU if we already cleaned something.
+            if (removed != 0)
+            {
+                return true;
+            }
+
+            if (oldUsed == DateTime.MaxValue)
+            {
+                // Something strange. Either capacity is 0 or all collections are locked with cc.Used.
+                return false;
+            }
+
+            // Remove oldest cookies from the least used collection.
+            lock (lruCc)
+            {
+                while (_count >= _maxCookies && lruCc.Count > 0)
+                {
+                    lruCc.RemoveAt(0);
+                    --_count;
+                }
+            }
+            return true;
+        }
+
+        // Return number of cookies removed from the collection.
+        private int ExpireCollection(CookieCollection cc)
+        {
+            lock (cc)
+            {
+                int oldCount = cc.Count;
+                int idx = oldCount - 1;
+
+                // Cannot use enumerator as we are going to alter collection.
+                while (idx >= 0)
+                {
+                    Cookie cookie = cc[idx];
+                    if (cookie.Expired)
+                    {
+                        cc.RemoveAt(idx);
+                    }
+                    --idx;
+                }
+                return oldCount - cc.Count;
+            }
+        }
+
+        public void Add(CookieCollection cookies)
+        {
+            if (cookies == null)
+            {
+                throw new ArgumentNullException("cookies");
+            }
+            foreach (Cookie c in cookies)
+            {
+                Add(c);
+            }
+        }
+
+        // This will try (if needed) get the full domain name of the host given the Uri.
+        // NEVER call this function from internal methods with 'fqdnRemote' == null.
+        // Since this method counts security issue for DNS and hence will slow
+        // the performance.
+        internal bool IsLocalDomain(string host)
+        {
+            int dot = host.IndexOf('.');
+            if (dot == -1)
+            {
+                // No choice but to treat it as a host on the local domain.
+                // This also covers 'localhost' and 'loopback'.
+                return true;
+            }
+
+            // Quick test for typical cases: loopback addresses for IPv4 and IPv6.
+            if ((host == "127.0.0.1") || (host == "::1") || (host == "0:0:0:0:0:0:0:1"))
+            {
+                return true;
+            }
+
+            // Test domain membership.
+            if (string.Compare(_fqdnMyDomain, 0, host, dot, _fqdnMyDomain.Length, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                return true;
+            }
+
+            // Test for "127.###.###.###" without using regex.
```

Comment:
Is the "without using regex" important because performance is important?  If so, we probably don't want to be using String.Split either, and instead just using IndexOf to walk each part on the original string.